libtock/tock.h

```diff
@@ -8,8 +8,65 @@
 extern "C" {
 #endif
 
-typedef void (subscribe_cb)(int, int, int,void*);
-
+typedef void (subscribe_cb)(int, int, int, void*);
+
+typedef enum {
+  TOCK_SYSCALL_FAILURE             =   0,
+  TOCK_SYSCALL_FAILURE_U32         =   1,
+  TOCK_SYSCALL_FAILURE_U32_U32     =   2,
+  TOCK_SYSCALL_FAILURE_U64         =   3,
+  TOCK_SYSCALL_SUCCESS             = 128,
+  TOCK_SYSCALL_SUCCESS_U32         = 129,
+  TOCK_SYSCALL_SUCCESS_U32_U32     = 130,
+  TOCK_SYSCALL_SUCCESS_U64         = 131,
+  TOCK_SYSCALL_SUCCESS_U32_U32_U32 = 132,
+  TOCK_SYSCALL_SUCCESS_U64_U32     = 133
+} syscall_rtype_t;
+
+typedef enum {
+  TOCK_ERROR_FAIL        = 0,
+  TOCK_ERROR_BUSY        = 1,
+  TOCK_ERROR_ALREADY     = 2,
+  TOCK_ERROR_OFF         = 3,
+  TOCK_ERROR_RESERVE     = 4,
+  TOCK_ERROR_INVAL       = 5,
+  TOCK_ERROR_SIZE        = 6,
+  TOCK_ERROR_CANCEL      = 7,
+  TOCK_ERROR_NOMEM       = 8,
+  TOCK_ERROR_NOSUPPORT   = 9,
+  TOCK_ERROR_NODEVICE    = 10,
+  TOCK_ERROR_UNINSTALLED = 11,
+  TOCK_ERROR_NOACK       = 12
+} tock_error_t;
+
+typedef struct {
+  syscall_rtype_t type;
+  uint32_t data[3];
+} syscall_return_t;
```

Comment:
Because this is a very general form; you could use it for any syscall, it represents the actual data passed back. Other syscalls have defined success and failure types, but command allows any (although a particular command has exactly one of each).

If I had inheritance, I'd make the other ones (subscribe, allow, etc) a subclass of this and just add accessors.